.github/workflows/ci.yml

```diff
@@ -0,0 +1,66 @@
+name: CI
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  # Many color libraries just need this to be set to any value, but at least
+  # one distinguishes color depth, where "3" -> "256-bit color".
+  FORCE_COLOR: 3
+  PROJECT_NAME: "MAWpy"
+
+jobs:
+  pre-commit:
+    name: Format
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+      - uses: pre-commit/action@v3.0.1
+        with:
+          extra_args: --hook-stage manual --all-files
+
+  checks:
```

Comment:
I see that the `pre-commit` job ran on this PR. Should this `checks` job have run too?

Reply:
Can we check if the "`Allow all actions and reusable workflows`" is selected for this repository. I do not have access to check this.

On my forked repository, this option is selected and I am able to see checks running when raising PR within that repository.

![image](https://github.com/uw-ssec/MAWpy/assets/21972901/d1906b6e-1e4d-408d-98d5-c4ce4fdcc0a1)

Reply:
This can be found at: `Settings --> Actions --> General --> Actions permissions`

Reply:
That setting is all set up. I think it's not running right now because it's from a fork. So once it's merged, it should run and subsequent changes would run then.